### PR TITLE
init: pnpm setup

### PR DESCRIPTION
**Pull request by danstarns**
# What?

This PR implements the following changes:

1. Adds [pnpm](https://pnpm.io/) for package management

It sets up the workspace as a monorepo containing two packages:

1. `grapesjs` (published)
2. `@grapesjs/docs` (non-published)

Both packages are managed through the root pnpm monorepo, including linting, shared dependencies, and build steps.

# Why?

We have a collection of repositories in our organization that depend on either the GrapesJS core or the GrapesJS CLI. Each repo currently contains slightly different versions and collections of issues. By moving these packages together, we aim to achieve:

- Better maintainability
- Higher visibility
- Faster time to ship

We chose [pnpm](https://pnpm.io/) because:

1. It's well-suited for managing many packages in a single repo (including potential future additions like component libraries, plugins, blocks, adapters, and tests).
2. It offers an excellent `workspaces:@` package resolution mechanism.

# Usage

> Note: pnpm is required globally.

For detailed usage instructions, please refer to the updated CONTRIBUTING.md file in this branch.

# Related

* https://github.com/GrapesJS/grapesjs/pull/6111

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -36,32 +36,21 @@
     "backbone-undo": "0.2.6",
     "codemirror": "5.63.0",
     "codemirror-formatting": "1.0.0",
+    "html-entities": "~1.4.0",
```
Comment (danstarns) [inline packages/core/package.json:38]: @artf this is needed to stop error: 

* https://github.com/vuejs/vuepress/issues/2996

Looks like we should keep it.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    "test": "pnpm --filter grapesjs test",
+    "docs": "pnpm --filter @grapesjs/docs docs",
+    "docs:api": "pnpm --filter @grapesjs/docs docs:api",
+    "lint": "eslint .",
+    "build": "pnpm --filter \"!@grapesjs/docs\" build",
+    "clean": "find . -type d \\( -name \"node_modules\" -o -name \"build\" -o -name \"dist\" \\) -exec rm -rf {} + && rm ./pnpm-lock.yaml",
+    "format": "prettier . --write",
+    "format:check": "prettier . --check"
+  },
+  "devDependencies": {
```
Comment (danstarns) [inline package.json:16]: Most of the dev deps for typescript, testing, building, and lining are now in this file.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "jest": "29.7.0",
+    "prettier": "3.3.3",
+    "ts-jest": "29.2.4",
+    "ts-loader": "9.5.1",
+    "ts-node": "10.9.2",
+    "typescript": "5.5.4"
+  },
+  "pnpm": {
+    "peerDependencyRules": {
+      "ignoreMissing": [
```
Comment (danstarns) [inline package.json:48]: They are hoisted to the root.

File: packages/core/package.json [hunk 4]
```diff
@@ -36,33 +35,18 @@
...
-    "@typescript-eslint/eslint-plugin": "8.2.0",
-    "@typescript-eslint/parser": "8.0.1",
-    "@vuepress/plugin-google-analytics": "1.8.2",
-    "@vuepress/types": "1.9.10",
-    "documentation": "14.0.3",
-    "eslint": "8.57.0",
-    "eslint-config-prettier": "9.1.0",
-    "eslint-plugin-jest": "28.8.0",
-    "eslint-plugin-prettier": "5.1.3",
     "grapesjs-cli": "4.1.3",
```
Comment (danstarns) [inline packages/core/package.json:44]: There is a conflict in the repo with `webpack`, as vuepress relies on a different version of webpack as to `grapesjs-cli`, we can resolve this better and share webpack when we have the cli in the monorepo.

File: docs/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "html-entities": "~1.4.0"
+  },
+  "devDependencies": {
+    "@types/markdown-it": "14.1.2",
+    "@vuepress/plugin-google-analytics": "1.8.2",
+    "@vuepress/types": "1.9.10",
+    "documentation": "14.0.3",
+    "postcss": "8",
+    "sass": "1.42.1",
+    "vuepress": "1.9.10",
```
Comment (danstarns) [inline docs/package.json:30]: Relies on webpack.

File: .github/workflows/quality.yml [hunk 6]
```diff
@@ -1,35 +1,23 @@
...
+name: GrapesJS Qualty Checks
 on:
   push:
     branches: [dev]
   pull_request:
     branches: [dev]
 
 jobs:
-  quality:
+  quailty-checks:
```
Comment (artf) [inline .github/workflows/quality.yml]: typo

File: CONTRIBUTING.md
Comment (artf) [inline CONTRIBUTING.md:1]: 💜

**Review by artf — APPROVED** [review]

**Pull request merged**